bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -525,9 +532,9 @@ public BookKeeper(ClientConfiguration conf, ZooKeeper zk, EventLoopGroup eventLo
         // initialize ledger manager
         try {
             this.ledgerManagerFactory =
-                AbstractZkLedgerManagerFactory.newLedgerManagerFactory(conf, regClient.getLayoutManager());
-        } catch (IOException | InterruptedException e) {
-            throw e;
+                this.metadataDriver.getLedgerManagerFactory();
```

Comment:
As with regClient. We don't need a member field for ledgerManagerFactory, as it is owned by the metadataDriver, so metadataDriver should clean up. Either that, or metadataDriver should be a pure factory and not hold a reference to ledgerManagerFactory and regClient, but as they likely rely on a shared resource (zk client?) it's probably better to just let metadataDriver own all.

Reply:
I am trying to reduce the amount code to change for places that references `ledgerManagerFactory`. this should be done via a cleanup later.

Reply:
In the latest patch, it is only ever referenced a couple of lines down in the constructor.

Reply:
there is a subsequent change on ledger manager factory. would defer changing that to ledger manager factory change, not sure it worth being stuck at here.

Reply:
for ledgerManagerFactory sure, I'd like regClient removed though.